src/app/components/Sidebar.tsx

```diff
@@ -0,0 +1,157 @@
+"use client";
+
+import React from "react";
+import {
+  useSidebarStateContext,
+  useSidebarActionsContext,
+} from "@context/SidebarContext";
+import {
+  Drawer,
+  Box,
+  Typography,
+  List,
+  ListItemButton,
+  Divider,
+} from "@mui/material";
+import theme from "theme";
+import { getCourses } from "@services/client/course";
+import { useUserSession } from "@context/UserSessionContext";
+import { useRouter } from "next/navigation";
+import { usePathname } from "next/navigation";
+import { IdentifiableCourse } from "@interfaces/type";
+import MenuButton from "./buttons/MenuButton";
+import ModeEditOutlinedIcon from "@mui/icons-material/ModeEditOutlined";
+
+const Sidebar = React.memo(() => {
```

Comment:
Why do we need both `Sidebar` and `SidebarContext`? Can the state not live inside this component?

Reply:
Ah I see it's because of the `Header` component.

Reply:
I thought about this too and I reasoned that:

- MUI Drawer component needs to in higher-order component as it covers the screen
- We would reduce prop drilling in the component tree 
- We would also reduce renders of Sidebar on different pages since it's static to userSession